lib/matplotlib/axes/_axes.py

```diff
@@ -5665,7 +5691,8 @@ def pcolormesh(self, *args, alpha=None, norm=None, cmap=None, vmin=None,
 
         allmatch = (shading == 'gouraud')
 
```

Comment:
```suggestion
dropdata = kwargs.pop('dropdata', True)

```
Also, the method definition for _pcolorargs need to be updated to include `dropdata=True`.

Reply:
What about a different kwarg than dropdata? Maybe something like xyvertices=False, xyedges. Something to insinuate that the current xy are grid points or vertices rather than edges.

Reply:
Yeah, I see what you mean, but not sure what to do about this.   We don't want to imply that we will do anything if `len(x)==N+1`.  I'll propose we keep `dropdata` for now.

Reply:
The problem I see with `dropdata` as the kwarg is that you're actually not **just** including all of the data, you're also changing the centers/edges of the cells by calculating a midpoint everywhere. In the examples above, the edges were at integers (0, 1, 2) and with this fix they are now at -0.5, 0.5, 1.5 2.5. So the grid has shifted as well. To me, `dropdata=False` reads like you would have just extended the previous edges to include a new extrapolation point to capture all of the data (0, 1, 2, 3).

xy_centers, xy_midpoints?

If `len(x)==N+1` and this kwarg is present, you could raise/warn a user that the combo doesn't make sense.

Reply:
> you're also changing the centers/edges of the cells by calculating a midpoint everywhere. 

I'm not calculating the midpoint, I'm calculating the edges, assuming that the given co-ordinates are the midpoint, which I think is what 99.99% of users would expect.  The fact is that I think this should be the default behaviour, but thats a whole back compatibility concern.

Reply:
I agree with what you've said (hence my `xy_midpoints` suggestion), I wasn't very clear in my description above. I also agree that I think this is what most users would expect. For backwards compatibility, I think it would be fine if this is just an extra option and the default is the previous behaviour.

Reply:
I agree that the name should be changed. I disagree that this is what users want 99% of the time. Most of the time, `imshow()` is exactly what those users really need, and is faster to boot. I have been in enough arguments with people about whether coordinates should represent vertices or midpoints to know that it is very dependent upon the particular field you are in and the standard data formats they are usually using most of the time.

Reply:
I think 0.001% of users expect data to be dropped.  As for imshow that is useless for unevenly spaced X or Y.